src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -6,30 +6,45 @@
 
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.CANSparkBase.IdleMode;
 import com.revrobotics.CANSparkLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.DigitalInput;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.HopperConstants;
 import frc.robot.Ports;
+import frc.robot.Ports.HopperPorts;
 
 public class Intake extends SubsystemBase {
-  /** Creates a new Intake. */
+  //intake
   // private static CANSparkMax rotationNEO;
   private static CANSparkMax rollerNEO;
   private static SparkAbsoluteEncoder encoder;
   // private static PIDController intakePIDController;
-  // private static boolean isRunning;
-  // private double setpoint;
+  private static boolean isRunning;
+  //private double setpoint;
+
+  //hopper
+  private final DigitalInput beamBreakIntake;
+  private final DigitalInput beamBreakShooter;
+  private final CANSparkMax hopperMotor;
 
   public Intake() {
+    //intake
     // rotationNEO = new CANSparkMax(Ports.IntakePorts.rotationNEOPort,
     // MotorType.kBrushless);
     rollerNEO = new CANSparkMax(Ports.IntakePorts.rollerNEOPort, MotorType.kBrushless);
-    // encoder = rotationNEO.getAbsoluteEncoder(Type.kDutyCycle);
-    // intakePIDController = new
-    // PIDController(Constants.IntakeConstants.PIDConstants.kP,
-    // Constants.IntakeConstants.PIDConstants.kI,
-    // Constants.IntakeConstants.PIDConstants.kD);
-    // setpoint = encoder.getPosition();
-    // isRunning = false;
+    //encoder = rotationNEO.getAbsoluteEncoder(Type.kDutyCycle); 
+    //intakePIDController = new PIDController(Constants.IntakeConstants.PIDConstants.kP, Constants.IntakeConstants.PIDConstants.kI, Constants.IntakeConstants.PIDConstants.kD);
+    //setpoint = encoder.getPosition();
+    isRunning = false;
+
+    //hopper
+    beamBreakIntake = new DigitalInput(HopperPorts.BEAM_BREAK_INTAKE_PORT);
+    beamBreakShooter = new DigitalInput(HopperPorts.BEAM_BREAK_SHOOTER_PORT);
```

Comment:
lines 43-44; refer to previous comment about naming beam breakers "receiver" and "emitter"